boskos/boskos.go

```diff
@@ -0,0 +1,383 @@
+			return
+		}
+
+		logrus.Infof("Request for a %v %v from %v", state, rtype, owner)
+
+		resource := r.Request(rtype, state, owner)
+
+		if resource != nil {
+			resJSON, err := json.Marshal(resource)
+			if err != nil {
+				logrus.WithError(err).Errorf("json.Marshal failed : %v", resource)
+				http.Error(res, err.Error(), http.StatusInternalServerError)
+				return
+			}
+			logrus.Infof("Resource leased: %v", string(resJSON))
+			fmt.Fprint(res, string(resJSON))
+		} else {
+			logrus.Infof("No available resource")
+			http.Error(res, "No resource", http.StatusInternalServerError)
+		}
+	}
+}
+
+func handleDone(r *Ranch) http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		logrus.WithField("handler", "handleDone").Infof("From %v", req.RemoteAddr)
+
+		if req.Method != "POST" {
+			logrus.Warning("[BadRequest]method %v, expect POST", req.Method)
+			http.Error(res, "/done only accept POST method", http.StatusBadRequest)
+			return
+		}
+
+		name := req.URL.Query().Get("name")
+		state := req.URL.Query().Get("state")
+		if name == "" || state == "" {
+			logrus.Warning("[BadRequest]name and state must present in the request")
+			http.Error(res, "Name and state must present in the request", http.StatusBadRequest)
+			return
+		}
+
+		if err := r.Done(name, state); err != nil {
+			logrus.WithError(err).Errorf("Done failed: %v - %v", name, state)
+			http.Error(res, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		logrus.Infof("Done with resource %v, set to state %v", name, state)
+	}
+}
+
+func handleReset(r *Ranch) http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		logrus.WithField("handler", "handleReset").Infof("From %v", req.RemoteAddr)
+
+		if req.Method != "POST" {
+			logrus.Warning("[BadRequest]method %v, expect POST", req.Method)
+			http.Error(res, "/reset only accept POST method", http.StatusBadRequest)
+			return
+		}
+
+		rtype := req.URL.Query().Get("type")
+		state := req.URL.Query().Get("state")
+		expireStr := req.URL.Query().Get("expire")
+		dest := req.URL.Query().Get("dest")
+
+		logrus.Infof("%v, %v, %v, %v", rtype, state, expireStr, dest)
+
+		if rtype == "" || state == "" || expireStr == "" || dest == "" {
+			logrus.Warning("[BadRequest]type, state, expire and dest must present in the request")
+			http.Error(res, "Type, state, expire and dest must present in the request", http.StatusBadRequest)
+			return
+		}
+
+		expire, err := time.ParseDuration(expireStr)
+		if err != nil {
+			logrus.WithError(err).Errorf("Invalid expire: %v", expireStr)
+			http.Error(res, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		rmap := r.Reset(rtype, state, expire, dest)
+		resJSON, err := json.Marshal(rmap)
+		if err != nil {
+			logrus.WithError(err).Errorf("json.Marshal failed : %v", rmap)
+			http.Error(res, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		logrus.Infof("Resource reset: %v", string(resJSON))
```

Comment:
Not sure we want to print the whole response. How about "reset <NUM> <TYPE> resources to <TYPE> after <DUR>"